fdbclient/MonitorLeader.actor.cpp

```diff
@@ -900,7 +907,23 @@ ACTOR Future<MonitorLeaderInfo> monitorProxiesOneGeneration(
 		state ClusterConnectionString storedConnectionString;
 		if (connRecord) {
 			bool upToDate = wait(connRecord->upToDate(storedConnectionString));
-			if (!upToDate) {
+			if (upToDate) {
+				incorrectTime = Optional<double>();
+			} else if (allConnectionsFailed) {
+				// Failed to connect to all coordinators from the current connection string,
+				// so it is not possible to get any new updates from the cluster. It can be that
+				// all the coordinators have changed, but the client missed that, because it had
+				// an incompatible protocol version. Since the cluster file is different,
+				// it may have been updated by other clients.
+				TraceEvent("UpdatingConnectionStringFromFile")
+				    .detail("ClusterFile", connRecord->toString())
+				    .detail("StoredConnectionString", storedConnectionString.toString())
+				    .detail("CurrentConnectionString", connRecord->getConnectionString().toString());
+				connRecord->setAndPersistConnectionString(storedConnectionString);
+				info.intermediateConnRecord =
```

Comment:
I think we can just set this directly, as `makeIntermediateRecord` is used to create a temporary copy of the connection record that isn't persisted. Since we've already updated and persisted it, we can just use it:

```suggestion
				info.intermediateConnRecord = connRecord;
```

Reply:
Makes sense. Thanks!